applications/vanilla/models/class.discussionmodel.php

```diff
@@ -716,11 +685,6 @@ public function calculate(&$Discussion) {
 
         $Discussion->CountCommentWatch = is_numeric($Discussion->CountCommentWatch) ? $Discussion->CountCommentWatch : null;
 
-        if ($Discussion->LastUserID == null) {
-            $Discussion->LastUserID = $Discussion->InsertUserID;
-            $Discussion->LastDate = $Discussion->DateInserted;
-        }
-
```

Comment:
This code is flawed and causes actual theme breakage right now (e.g. see vanillaforums.org/discussion/30496/issue-recent-discussions-module-doesnt-show-author).

That's what this PR is all about - remove these incomplete and actually already theme-breaking legacy calculated columns and adapt old themes to no longer rely on their presence.